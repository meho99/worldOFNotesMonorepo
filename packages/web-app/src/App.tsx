import React from 'react'

export const App = () => {
  return (
    <div>Test</div>
  )
}
App.displayName = 'AppComponent'
