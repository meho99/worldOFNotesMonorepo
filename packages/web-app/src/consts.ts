export enum keyNames {
  Session = 'session',
  Notes = 'notes',
  Router = 'router'
} 
