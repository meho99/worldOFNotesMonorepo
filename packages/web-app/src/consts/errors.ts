export enum FieldsErrors {
  PasswordToShort = 'Password must have at least 8 characters',
  PasswordToLong = 'Password must have less than 8 characters',
  NameToShort = 'Name must have at least 3 characters',
  NameToLong = 'Name must have less than 25 characters',
  Required = 'Required field',
  InvalidEmailFormat = 'Invalid email address'
}

export enum FetchingErrors {
  LoginError = 'Unable to login. Please check your email and password and try again.',
  SingUpError = 'Unable to sign up. Please check your email and password and try again.',
  Unauthorized = 'Unauthorized'
}
