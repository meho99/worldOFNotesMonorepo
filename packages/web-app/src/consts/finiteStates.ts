export enum FiniteStates {
  Idle = 'idle',
  Loading = 'loading',
  Success = 'success',
  Failure = 'failure',
}
