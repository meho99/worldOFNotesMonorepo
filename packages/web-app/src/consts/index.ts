export * from './urls'
export * from './themeTypes'
export * from './finiteStates'
export * from './reducerNames'
export * from './errors'
export * from './notificationTypes'
