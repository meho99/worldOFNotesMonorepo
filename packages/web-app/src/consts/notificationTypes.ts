export enum NotificationTypes {
  Error = 'error'
}
