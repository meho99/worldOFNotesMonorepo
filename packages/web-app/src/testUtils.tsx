import React from 'react'
import { DeepPartial } from 'redux'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { ReducerState } from './redux/reducers'

const mockStore = configureStore()

export const CreateProvider = (testState: DeepPartial<ReducerState>): React.FC => ({ children }) =>
  <Provider store={mockStore(testState)}> {children} </Provider>
