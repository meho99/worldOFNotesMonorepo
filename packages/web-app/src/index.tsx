import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { createGenerateClassName, StylesProvider } from '@material-ui/core/styles'

import { store } from './redux'
import { ThemeProvider } from './theme'

import { App } from './App'

const generateClassName = createGenerateClassName({
  productionPrefix: 'won',
  disableGlobal: true
})

ReactDOM.render(
  <StylesProvider generateClassName={generateClassName} injectFirst>
    <ThemeProvider variant='dark'>
      <Provider store={store}>
        <App />
      </Provider>
    </ThemeProvider>
  </StylesProvider>,
  document.getElementById('root')
)
