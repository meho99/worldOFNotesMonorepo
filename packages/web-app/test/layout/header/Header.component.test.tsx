import React from 'react'
import { render } from '@testing-library/react'

import { HeaderComponent } from '../../../src/layout/header/Header.component'
import { FiniteStates } from '../../../src/consts'
import { CreateProvider } from '../../../src/testUtils'

describe('HeaderComponent test', () => {
  it('Snapshot test', () => {
    expect(
      render(<HeaderComponent />, {
        wrapper: CreateProvider({
          session: {
            authenticatingStatus: FiniteStates.Success,
          },
        }),
      }).container,
    ).toMatchSnapshot()
  })
})
