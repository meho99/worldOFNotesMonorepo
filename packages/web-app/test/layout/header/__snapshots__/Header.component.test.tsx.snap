// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent test Snapshot test 1`] = `
<div>
   
   
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionFixed mui-fixed css-7kuu8v-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 makeStyles-title-2 css-2ulfj5-MuiTypography-root"
      >
        World of Notes
      </h6>
      <div>
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium makeStyles-menuButton-1 css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
   
</div>
`;
